src/snr_measurements.py

```diff
@@ -305,8 +306,163 @@ def check_xt(self):
             resp = cross_talk < self.engine_props['requested_xt'][self.spectrum_props.modulation]
 
         return resp, cross_talk
+    
+    # TODO: update the method based on external resources
+    def check_snr_ext(self, path_index):
+        """
+        Checks the SNR on a single request using the external resources.
+
+        :return: Whether the SNR threshold can be met and SNR value.
+        :rtype: tuple
+        """
+        mod_format_mapping = {
+        6: "64-QAM",
+        5: "32-QAM",
+        4: "16-QAM",
+        3: "8-QAM",
+        2: "QPSK",
+        1: "BPSK"
+        }
+        BW_mapping = {
+        "64-QAM": 600,
+        "32-QAM": 500,
+        "16-QAM": 400,
+        "8-QAM": 300,
+        "QPSK": 200,
+        "BPSK": 100
+        }
+        if self.engine_props['multi_fiber']:
+            loaded_data = np.load('MF-USB6014-MF.npy', allow_pickle=True)
+            loaded_data_gsnr = np.load('GSNR-USB6014-MF.npy', allow_pickle=True)
+        else:
+            if self.engine_props['cores_per_link'] == 4:
```

Comment:
A function in "snr_helpers" would help the organization of this method a lot. Or, have another argument in "snr_args" where it's a tuple mapping to file names. For example: (core_num, cores_per_link): [file1, file2]. The function could then do something like (You'd need to account for a None type, for example the first if statement only cares about cores_per_link):

`file_names_list = snr_args[cores_per_link, core_num]
file_one, file_two = np.load()...
`